src/transformers/models/decision_transformer/modeling_decision_transformer.py

```diff
@@ -178,7 +178,9 @@ def _attn(self, query, key, value, attention_mask=None, head_mask=None):
         attn_weights = torch.matmul(query, key.transpose(-1, -2))
 
         if self.scale_attn_weights:
-            attn_weights = attn_weights / (value.size(-1) ** 0.5)
+            attn_weights = attn_weights / torch.tensor(value.size(-1) ** 0.5, dtype=attn_weights.dtype).to(
+                attn_weights.device
+            )
```

Comment:
Nit, how about something like this @JingyaHuang ?

Other than that, and the same below, happy to merge it

```suggestion
            attn_weights = attn_weights / torch.tensor(value.size(-1) ** 0.5, dtype=attn_weights.dtype, device=attn_weights.device)
```

Reply:
Fixed for both, thanks for the suggestion!